src/Symfony/Component/Messenger/Middleware/AllowSingleHandlerMiddleware.php

```diff
@@ -0,0 +1,43 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Messenger\Middleware;
+
+use Symfony\Component\Messenger\Exception\MoreThanOneHandlerForMessageException;
+use Symfony\Component\Messenger\Handler\ChainHandler;
+use Symfony\Component\Messenger\Handler\Locator\HandlerLocatorInterface;
+
+/**
+ * @author Kamil Kokot <kamil@kokot.me>
+ */
+class AllowSingleHandlerMiddleware implements MiddlewareInterface
+{
+    private $messageHandlerResolver;
+
+    public function __construct(HandlerLocatorInterface $messageHandlerResolver)
+    {
+        $this->messageHandlerResolver = $messageHandlerResolver;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function handle($message, callable $next)
+    {
+        $handler = $this->messageHandlerResolver->resolve($message);
+
+        if ($handler instanceof ChainHandler) {
```

Comment:
I feel like this is tight coupling to a specific implementation: any other handler could implement a chained behavior, isn't it?
Why do we want to restrict the possibilities here?

Reply:
Yes, this is coupled with the `FrameworkBundle` implementation. Unfortunately, I don't have any idea on how to make it better (I don't think that sharing the YAML configuration is better either). @nicolas-grekas do you have an idea? 🤔